src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -379,8 +379,14 @@ public async Task<Index> CreateIndex(IndexDefinition definition)
 
             var instance = IndexCompilationCache.GetIndexInstance(definition, _documentDatabase.Configuration); // pre-compile it and validate
 
+            if (NeedToUpdateIndex(definition, out Index currentIndex, out var validateMapReduceCollectionName) == false)
```

Comment:
Why do we need to handle that here? Based on the included test I understand the issue was that we didn't recognize the index definition has _not_ been changed meanwhile, right?

Why we didn't filter it out here?:
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Client/ServerWide/DatabaseRecord.cs#L93

Reply:
This is before we try to add it to the database record